File: Python/main62.py
# date and time

import datetime

date = datetime.date(2025, 1, 13)
today = datetime.date.today()

time = datetime.time(12, 30, 0)
now = datetime.datetime.now()

now = now.strftime("%H:%M:%S %d-%m-%Y")

print(date)
print(today)

print(time)
print(now)


target_datetime = datetime.datetime(2030, 1, 2, 12, 30, 1)
target_datetime1 = datetime.datetime(2021, 2, 3, 12, 3, 4)
current_datetime = datetime.datetime.now()

if target_datetime < current_datetime:
    print("Target date has passed")
else:
    print("Target date has NOT passed")

if target_datetime1 < current_datetime:
    print("Target date has passed")
else:
    print("Target date has NOT passed")

File: JAVA/Main62.java
package JAVA;

import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.time.format.DateTimeFormatter;

public class Main62{
    public static void main(String[] args){

        // How to work with DATES & TIMES using JAVA
        // (LocalDate, LocalTime, LocalDateTime, UTC timestamp)

        LocalDate date = LocalDate.now();
        LocalTime time = LocalTime.now();
        LocalDateTime dateTime = LocalDateTime.now();
        Instant instant  = Instant.now();

        System.out.println(date);
        System.out.println(time);
        System.out.println(dateTime);
        System.out.println(instant);


        // Custom format

        LocalDateTime dateTime1 = LocalDateTime.now();

        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm:ss");
        String newDateTime = dateTime1.format(formatter);
        System.out.println(newDateTime);

        LocalDateTime date1 = LocalDateTime.of(1267, 4, 17, 5, 7, 2);
        LocalDateTime date2 = LocalDateTime.of(2025, 4, 17, 5, 7, 34);

        System.out.println(date1);
        System.out.println(date2);


        if (date1.isBefore(date2)){
            System.out.println(date1 + " is earlier than " + date2);
        }
        else if (date1.isBefore(date2)){
            System.out.println(date1 + " is later than " + date2);
        }
        else{
            System.out.println(date1 + " is equal to " + date2);
        }
    }
}